lib/lndclient/LndClient.ts

```diff
@@ -295,84 +295,64 @@ class LndClient extends SwapClient {
-          throw new Error(sendPaymentError);
-        }
-
-        return base64ToHex(sendToRouteResponse.getPaymentPreimage_asB64());
-      } catch (err) {
-        this.logger.error(`got exception from sendToRouteSync: ${JSON.stringify(request.toObject())}`, err);
-        throw err;
+    let request: lndrpc.SendRequest;
+    if (deal.role === SwapRole.Taker) {
+      // we are the taker paying the maker
+      if (!deal.destination) {
+        assert.fail('swap deal as taker must have a destination');
       }
-    } else if (deal.destination) {
-      const request = new lndrpc.SendRequest();
-      request.setDestString(deal.destination);
-      request.setPaymentHashString(deal.rHash);
-
-      if (deal.role === SwapRole.Taker) {
-        // we are the taker paying the maker
-        request.setFinalCltvDelta(deal.makerCltvDelta!);
-        request.setAmt(deal.makerUnits);
-      } else {
-        // we are the maker paying the taker
-        request.setFinalCltvDelta(deal.takerCltvDelta);
-        request.setAmt(deal.takerUnits);
-      }
-      const fee = new lndrpc.FeeLimit();
-      fee.setFixed(Math.floor(MAXFEE * request.getAmt()));
-      request.setFeeLimit(fee);
-
-      try {
-        this.logger.debug(`executing sendPaymentSync -
-          destination: ${deal.destination}, rHash: ${deal.rHash}, amount: ${request.getAmt()}, FinalCltvDelta: ${request.getFinalCltvDelta()}`);
-        const sendPaymentResponse = await this.sendPaymentSync(request);
-        const sendPaymentError = sendPaymentResponse.getPaymentError();
-        if (sendPaymentError) {
-          this.logger.error(`sendPaymentSync failed with payment error: ${sendPaymentError}`);
-          throw new Error(sendPaymentError);
-        }
-
-        return base64ToHex(sendPaymentResponse.getPaymentPreimage_asB64());
-      } catch (err) {
-        this.logger.error(`got exception from sendPaymentSync: ${JSON.stringify(request.toObject())}`, err);
-        throw err;
+      if (!deal.makerCltvDelta) {
+        assert.fail('swap deal as taker must have a makerCltvDelta');
       }
+      request = this.buildSendRequest({
+        rHash: deal.rHash,
+        destination: deal.destination!,
+        amount: deal.makerAmount,
+        // Using the agreed upon makerCltvDelta. Maker won't accept
+        // our payment if we provide a smaller value.
+        finalCltvDelta: deal.makerCltvDelta!,
+      });
     } else {
-      assert.fail('swap deal must have a route or destination to send payment');
-      return '';
+      // we are the maker paying the taker
+      if (!deal.takerPubKey) {
+        assert.fail('swap deal as maker must have a takerPubKey');
+      }
+      if (!deal.takerCltvDelta) {
+        assert.fail('swap deal as maker must have a takerCltvDelta');
+      }
+      if (!deal.makerCltvDelta) {
+        assert.fail('swap deal as maker must have a makerCltvDelta');
```

Comment:
All these checks can be on one line like `assert(deal.makerCltvDelta, 'swap deal as maker must have a makerCltvDelta')` - no need to have the if statements. I think it'd be a bit more readable.